### DISCUSSION TITLE
Evaluating Uno for WCAG 2.1 AA Compliance: Insights and Tips?

### DISCUSSION THREAD

**Discussion by lacy-skrz** [unoplatform/uno#18977] [General]
I’m currently evaluating the Uno Platform to determine if it can meet our accessibility requirements for a complex application, specifically compliance with WCAG 2.1 Level AA. Details about this technical standard for mobile apps can be found on the ADA's [Fact Sheet](https://www.ada.gov/resources/2024-03-08-web-rule/). 

From Uno's [documentation](https://platform.uno/docs/articles/features/working-with-accessibility.html), it’s clear that some important needs are addressed, like responsive design, text scaling, and integration with assistive technologies, especially mobile screen readers. However, I’ve noticed potential gaps, such as limited support for UI AutomationProperties/Automation Peers, and specific issues like the absence of TabIndex functionality.

I’m curious if anyone here has experience creating accessible applications with Uno, particularly meeting WCAG technical standards. Could you share insights on:

1. Any workarounds or solutions for missing features like TabIndex and unsupported AutomationProperties?
2. Overall capability to create an accessible experience that adheres to WCAG standards?

Thanks in advance for your help—I appreciate any input from those with experience navigating these challenges!

**Comment by MartinZikmund** [comment]
Thank you for considering Uno Platform! Indeed our a11y support is a work in progress. In terms of automation support we currently have some improvements in the works which will be published around beginning of new year. In addition, `TabIndex` is now fully supported, that is in the current stable version of Uno Platform already (we just failed to update the docs here...)

I think the best approach would be to see which specific APIs/functionalities is Uno Platform lacking when compared to WinUI 3 directly, and creating separate issue for those - we can then start prioritizing the missing features accordingly.

**Comment by sasakrsmanovic** [comment]
Happy to get on a call if this will be faster @lacy-skrz . Just reach out to info at platform.uno with your availability.

**Comment by francoistanguay** [comment]
We had partner agencies developer several WCAG-compliant apps with considerations on Contrast/Zoom/Font Size and others.

As Sasha suggested, meeting to better understand specific requirements is always a great way to help us figure out how to fill some of the minor gaps that might be left.

**Comment by sasakrsmanovic** [comment]
Hello @lacy-skrz ; were the answers above sufficient? Let us know please.
